src/components/common/Button/Button.tsx

```diff
@@ -26,18 +26,16 @@ const shadeColor = (color, percent) => {
 const colorUnderButtons = (hexColor) => {
   const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hexColor);
   return result
-    ? `rgba(${parseInt(result[1], 16) - 102},${parseInt(result[2], 16) - 55},${
-        parseInt(result[3], 16) - 14
-      },1)`
+    ? `rgba(${parseInt(result[1], 16) - 102},${parseInt(result[2], 16) - 55},${parseInt(result[3], 16) - 14
+    },1)`
     : null;
 };
 
 const onHoverBackgroundColor = (hexColor) => {
   const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hexColor);
   return result
-    ? `rgb(${parseInt(result[1], 16) + 44},${parseInt(result[2], 16) + 44},${
-        parseInt(result[3], 16) + 44
-      })`
+    ? `rgb(${parseInt(result[1], 16) + 44},${parseInt(result[2], 16) + 44},${parseInt(result[3], 16) + 44
+    })`
     : null;
 };
 const Button = styled.a`
```

Comment:
Can we handle types of props for this styled components?
We want to acheive something similar to this: https://styled-components.com/docs/api#using-custom-props

Reply:
![image](https://user-images.githubusercontent.com/36299161/144629136-de6b60a6-be27-4a0d-8f84-6b020f34299d.png)

Adding types to the styledcomponent as suggested in the link provided does not enforce types on the function arguments in the code , I will have to assign types to every function as seen on line 58 in the image

Reply:
what is <ButtonProps> type first of all?

Reply:
interface ButtonProps {
  bgColor: string
  textColor: string
}

Reply:
okay that's good because every time you use that component you can't pass prop bgColor of object (bgColor always has to be of type string).

Reply:
> 

Although I would not use those as mandatory props, so add question mark before the colon, 
here's an example https://www.typescriptlang.org/docs/handbook/2/objects.html#optional-properties

Reply:
Okay, but is it not meant to affect the argument in the functions?

If I hover over marginTop on line 62, it shows type: any ... even after setting it to type string in the interface ButtonProps

Reply:
it's weird that it's showing Any in the arguments but as long as it shows the appropriate props when you use the component 
then it's fine.